app/models/miq_worker.rb

```diff
@@ -514,7 +514,7 @@ def minimal_class_name
 
   def database_application_name
     zone = MiqServer.my_server.zone
-    "MIQ #{Process.pid} #{minimal_class_name}[#{compressed_id}], s[#{miq_server.compressed_id}], #{zone.name}[#{zone.compressed_id}]".truncate(64)
+    "MIQ|#{Process.pid}|#{miq_server.compressed_id}|#{compressed_id}|#{zone.compressed_id}|#{minimal_class_name}|#{zone.name}".truncate(64)
   end
 
```

Comment:
Is this to find if it's truncated? If so, we can directly test application_name to see if its truncated. If it's truncated, it's guaranteed to end with `...` if passing no additional options to `truncate`
https://github.com/rails/rails/blob/master/activesupport/lib/active_support/core_ext/string/filters.rb#L47